chef/cookbooks/ohai/recipes/default.rb

```diff
@@ -54,27 +54,29 @@
 end
 rd.run_action(:create)
 
-# we need to ensure that the cstruct gem is available (since we use it in our
-# plugin), except on sledgehammer (because it's already installed and we can't
-# install/check packages there)
-states = [ "ready", "readying", "recovering", "applying" ]
-if states.include?(node[:state])
-  # During the upgrade process (stoney -> tex, old ruby&rails -> tex
-  # ruby&rails), we need to run the new cookbook with the old ruby&rails once,
-  # so we need to support this
-  if node["languages"]["ruby"]["version"].to_f == 1.8
-    pkg = "rubygem-cstruct"
-  else
-    pkg = "ruby#{node["languages"]["ruby"]["version"].to_f}-rubygem-cstruct"
-  end
-  package(pkg).run_action(:install)
+unless node[:platform] == "windows"
```

Comment:
```ruby
if node[:platform] == "windows"
  chef_gem "cstruct"
else
```

On the case of ovisan the cstruct gem was missing on his hyperv machine. That way we install it in the chef scope as windows got chef 11 installed.